src/moscot/backends/ott/solver.py

```diff
@@ -88,17 +93,62 @@ def _create_geometry(
             return geometry.Geometry(
                 kernel_matrix=arr, epsilon=epsilon, relative_epsilon=relative_epsilon, scale_cost=scale_cost
             )
+        if x.is_graph:  # we currently only support this for the linear term.
+            return self._create_graph_geometry(
+                x=x, arr=arr, problem_kind=self.problem_kind, problem_shape=problem_shape, t=t, **kwargs  # type: ignore[arg-type]
+            )
         raise NotImplementedError(f"Creating geometry from `tag={x.tag!r}` is not yet implemented.")
 
     def _solve(  # type: ignore[override]
         self,
         prob: OTTProblem_t,
         **kwargs: Any,
-    ) -> OTTOutput:
+    ) -> Union[OTTOutput, GraphOTTOutput]:
         solver = jax.jit(self.solver) if self._jit else self.solver
         out = solver(prob, **kwargs)
+        if isinstance(prob, linear_problem.LinearProblem) and isinstance(prob.geom_xy, geodesic.Geodesic):
+            return GraphOTTOutput(out, shape=(len(self._a), len(self._b)))  # type: ignore[arg-type]
         return OTTOutput(out)
 
+    def _create_graph_geometry(
+        self,
+        x: TaggedArray,
+        arr: jax.Array,
+        problem_shape: Tuple[int, int],
+        t: Optional[float],
+        epsilon: Union[float, epsilon_scheduler.Epsilon] = None,
+        relative_epsilon: Optional[bool] = None,
+        scale_cost: Scale_t = 1.0,
+        **kwargs: Any,
+    ) -> geometry.Geometry:
+        if x.cost == "geodesic":
+            if self.problem_kind == "linear":
+                if t is None:
+                    return geodesic.Geodesic.from_graph(
+                        arr, t=epsilon / 4.0, directed=kwargs.pop("directed", True), **kwargs
+                    )
+
+                n_src, n_tgt = problem_shape
+                if n_src + n_tgt != arr.shape[0]:
+                    raise ValueError(f"Expected `x` to have `{n_src + n_tgt}` points, found `{arr.shape[0]}`.")
+                cm = geodesic.Geodesic.from_graph(
+                    arr, t=t, directed=kwargs.pop("directed", True), **kwargs
+                ).cost_matrix[:n_src, n_src:]
```

Comment:
why do you instantiate cost? it's not in the quadratic case right?

Reply:
actually the whole bit above. I don't see why it's needed? the quadratic case is below

Reply:
If `t!=eps/4` we have to instantiate it.